File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cs
public class Solution {
    public int[] SearchRange(int[] nums, int target) {
     int[] startEnd = {-1,-1};
     int left = 0;
     int right = nums.Length -1;
     int mid = (left + right)/2;    
     while(left<=right){
         mid = (left + right)/2; 
         if(nums[mid]<target){
             left = mid +1;
         }else if(nums[mid]>target){
             right = mid -1;
         }else{
             
             left = right+1;
             bool terminateStart = false;
             bool terminateEnd = false;
             startEnd[0]=mid;
             startEnd[1]=mid;
             
             while(!(terminateStart&& terminateEnd)){
             if(startEnd[0]-1>=0 && nums[startEnd[0]-1]==nums[mid]){
                
                 startEnd[0]-=1;
             }else{
                 terminateStart = true;
             }
             if(startEnd[1]+1<nums.Length && nums[startEnd[1]+1]==nums[mid]){
                 startEnd[1]+=1;
                 
             }else{
                 terminateEnd = true;
             }
                 
             
         }
         
     }
     }
        return startEnd;
    }
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        start, end = 0, len(nums) - 1
        
        ans = [-1,-1]
        mid = 0
        if len(nums) < 1: return ans
        
        while start <= end:
            mid = (start + end)//2
            
            if nums[mid] < target:
                start = mid + 1
            else:
                end = mid - 1
                ans[0] = mid 
        
        start, end = 0, len(nums) - 1
        while start <= end:
            mid = (start + end)//2
            
            if nums[mid] > target:
                end = mid - 1
            else:
                start = mid + 1
                ans[1] = mid
        if ans[0] > ans[1] or ans[0] < 0 or ans[1] < 0 or ans[1] >= len(nums) or ans[0] >= len(nums): return [-1,-1]
        return ans